Automattic/jetpack
# Sync: Don't sync set object terms action for blacklisted taxonomies

Fixes https://github.com/Automattic/vulcan/issues/530

As explained in the issue above, even if the action is sent, the processing in WPCOM will not store the term relationship since the blacklisted taxonomy never gets registered.
## Proposed changes:

* Check if the taxonomy is blacklisted prior to send the `set_object_terms` action.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






In a Jetpack connected site, let's leverage the wp shell (If running docker locally, you can run `jetpack docker wp shell`). 
We will do the same steps for trunk and this branch.
- Let's first register a taxonomy that is blacklisted (e.g ´widget`) with posts
    - `register_taxonomy('widget', array( 'post' ), array())`
- Insert a term for the taxonomy. Normally a  `jetpack_sync_add_term` action will be triggered for this but since the taxonomy is blacklisted it will not be sent either in trunk or in this PR.
    - ` wp_insert_term('my_widget', 'widget')` 
-  Set the object `term_id` returned in prior step to a `post_id` in your local site. It should return an array with the `term_id`. In trunk the `set_object_terms` action will be sent, with the PR applied action should not be sent.
    - `wp_set_object_terms(post_id,term_id, 'widget')` 

**Note:** To check if actions are being send or not, best way is by checking the ES logs. 

For extra points, test the above steps with a non-blacklisted taxonomy. The `jetpack_sync_add_term` and `set_object_terms` should be sent for both trunk and this PR
